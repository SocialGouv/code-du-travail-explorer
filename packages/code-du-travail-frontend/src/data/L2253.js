// https://www.legifrance.gouv.fr/affichCodeArticle.do?cidTexte=LEGITEXT000006072050&idArticle=LEGIARTI000006901774
export const THEMES_L22531 = {
  "salaires-minima-hierarchiques": [113500],
  classifications: [],
  "mutualisation-fonds-financement-paritarisme": [97200],
  "mutualisation-fonds-formation-professionnelle": [68200],
  "garanties-collectives-complementaires": [],
  "mesures-duree-travail": [20500, 19600, 21600, 21500],
  "mesures-cdd-ctt": [49500, 50500, 50900, 51900, 52200, 52600],
  "mesures-cdi-chantier": [12800],
  "egalite-professionnelle": [102200],
  "conditions-durees-renouvellement-periode-essai": [3700,3710,3720,3730,3740,3750,3760,3770],
  "modalites-poursuite-contrats-travail": [12100],
  "mise-a-disposition-salarie-temporaire": [51400],
  "remuneration-minimale-salarie": [44400]
};

// https://www.legifrance.gouv.fr/affichCodeArticle.do?cidTexte=LEGITEXT000006072050&idArticle=LEGIARTI000036262246&dateTexte=&categorieLien=id
export const THEMES_L22532 = {
  "prevention-exposition-facteurs-risques-professionnels": [162100],
  "insertion-professionnelle-maintien-emploi-handicapes": [86500],
  "effectif-delegues-syndicaux": [165200],
  "prime-travaux-dangereux-insalubres": [41100]
};
