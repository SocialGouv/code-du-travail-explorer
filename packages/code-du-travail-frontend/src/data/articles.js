// mettre les articles dont nous avons besoin
export default {
  "R3111-1": {
    texte:
      "A l'exception du chapitre II du titre III ainsi que des titres VI et VII, le présent livre définit les règles d'ordre public, le champ de la négociation collective et les règles supplétives applicables en l'absence d'accord."
  },
  "L3111-1": {
    texte:
      "Les dispositions du présent livre sont applicables aux employeurs de droit privé ainsi qu'à leurs salariés.   Elles sont également applicables aux établissements publics à caractère industriel et commercial."
  },
  //période d'essai
  "L1221-19": {
    texte:
      "Le contrat de travail à durée indéterminée peut comporter une période d'essai dont la durée maximale est :<br><br> 1° Pour les ouvriers et les employés, de deux mois ; <br><br>2° Pour les agents de maîtrise et les techniciens, de trois mois ; <br><br>3° Pour les cadres, de quatre mois."
  },
  "L1221-20": {
    texte:
      "La période d'essai permet à l'employeur d'évaluer les compétences du salarié dans son travail, notamment au regard de son expérience, et au salarié d'apprécier si les fonctions occupées lui conviennent. "
  },
  "L1221-21": {
    texte:
      "La période d'essai peut être renouvelée une fois si un accord de branche étendu le prévoit. Cet accord fixe les conditions et les durées de renouvellement. La durée de la période d'essai, renouvellement compris, ne peut pas dépasser : 1° Quatre mois pour les ouvriers et employés ; 2° Six mois pour les agents de maîtrise et techniciens ; 3° Huit mois pour les cadres."
  },
  "L1221-22": {
    texte:
      "Les durées des périodes d'essai fixées par les articles L. 1221-19 et L. 1221-21 ont un caractère impératif, à l'exception :de durées plus longues fixées par les accords de branche conclus avant la date de publication de la loi n° 2008-596 du 25 juin 2008 portant modernisation du marché du travail ; de durées plus courtes fixées par des accords collectifs conclus après la date de publication de la loi n° 2008-596 du 25 juin 2008 précitée ;- de durées plus courtes fixées dans la lettre d'engagement ou le contrat de travail."
  },
  "L1221-23": {
    texte:
      "La période d'essai et la possibilité de la renouveler ne se présument pas. Elles sont expressément stipulées dans la lettre d'engagement ou le contrat de travail"

    //durée max quotidienne
  },
  "L3121-18": {
    texte:
      "<p>La durée quotidienne de travail effectif par salarié ne peut excéder dix heures, sauf :</p><p>1° En cas de dérogation accordée par l'inspecteur du travail dans des conditions déterminées par décret ;</p><p>2° En cas d'urgence, dans des conditions déterminées par décret ;</p><p>3° Dans les cas prévus à l'article L. 3121-19.</p>"
  },
  "L3121-19": {
    texte:
      "Une convention ou un accord d'entreprise ou d'établissement ou, à défaut, une convention ou un accord de branche peut prévoir le dépassement de la durée maximale quotidienne de travail effectif, en cas d'activité accrue ou pour des motifs liés à l'organisation de l'entreprise, à condition que ce dépassement n'ait pas pour effet de porter cette durée à plus de douze heures."
  },
  "D3121-4": {
    texte:
      "<p>Le dépassement de la durée quotidienne maximale du travail effectif, prévue à l'article L. 3121-18, peut être autorisé dans les cas où un surcroît temporaire d'activité est imposé, notamment pour l'un des motifs suivants :</p><p>1° Travaux devant être exécutés dans un délai déterminé en raison de leur nature, des charges imposées à l'entreprise ou des engagements contractés par celle-ci ;</p><p>2° Travaux saisonniers ;</p><p>3° Travaux impliquant une activité accrue pendant certains jours de la semaine, du mois ou de l'année.</p>"
  },
  "D3121-5": {
    texte:
      "<p>La demande de dépassement de la durée quotidienne maximale de travail, accompagnée des justifications utiles et de l'avis du comité social et économique, s'il existe, est adressée par l'employeur à l'inspecteur du travail.</p><p>L'inspecteur du travail fait connaître sa décision dans un délai de quinze jours à compter de la date de réception de la demande à l'employeur et aux représentants du personnel.</p>"
  },
  "D3121-6": {
    texte:
      "<p>En cas d'urgence, l'employeur peut dépasser sous sa propre responsabilité, dans les hypothèses envisagées l'article D. 3121-4, à la durée quotidienne maximale du travail.</p><p>S'il n'a pas encore adressé de demande de dépassement, il présente immédiatement à l'inspecteur du travail une demande de régularisation accompagnée des justifications et avis mentionnés à l'article D. 3121-5 et de toutes explications nécessaires sur les causes ayant nécessité une prolongation de la durée quotidienne du travail sans autorisation préalable.</p><p>S'il se trouve dans l'attente d'une réponse à une demande de dépassement, il informe immédiatement l'inspecteur du travail de l'obligation où il s'est trouvé d'anticiper la décision attendue et en donne les raisons.</p><p>L'inspecteur du travail fait connaître sa décision dans un délai de quinze jours à compter de la date de réception de la demande à l'employeur et aux représentants du personnel.</p>"
  },
  "R3124-3": {
    texte:
      "Le fait de méconnaître les dispositions relatives à la durée légale hebdomadaire et à la durée quotidienne maximale du travail prévues par les articles L. 3121-10 et L. 3121-34 ainsi que celles des décrets prévus par les articles L. 3121-52 et L. 3122-46, est puni de l'amende prévue pour les contraventions de la quatrième classe. Les contraventions donnent lieu à autant d'amendes qu'il y a de salariés indûment employés."
  },
  "D3121-7": {
    texte:
      "Les recours hiérarchiques contre les décisions prévues aux articles D. 3121-5 et D. 3121-6 sont formés devant le directeur régional des entreprises, de la concurrence, de la consommation, du travail et de l'emploi dans le délai d'un mois suivant la date à laquelle les intéressés en ont reçu notification."
  },

  // RC

  "L1237-11": {
    texte:
      "<p>L'employeur et le salarié peuvent convenir en commun des conditions de la rupture du contrat de travail qui les lie.</p><p>La rupture conventionnelle, exclusive du licenciement ou de la démission, ne peut être imposée par l'une ou l'autre des parties.</p><p>Elle résulte d'une convention signée par les parties au contrat. Elle est soumise aux dispositions de la présente section destinées à garantir la liberté du consentement des parties.</p>"
  },
  "L1237-12": {
    texte:
      "<p>Les parties au contrat conviennent du principe d'une rupture conventionnelle lors d'un ou plusieurs entretiens au cours desquels le salarié peut se faire assister :<br>1° Soit par une personne de son choix appartenant au personnel de l'entreprise, qu'il s'agisse d'un salarié titulaire d'un mandat syndical ou d'un salarié membre d'une institution représentative du personnel ou tout autre salarié ;<br>2° Soit, en l'absence d'institution représentative du personnel dans l'entreprise, par un conseiller du salarié choisi sur une liste dressée par l'autorité administrative.</br><p>Lors du ou des entretiens, l'employeur a la faculté de se faire assister quand le salarié en fait lui-même usage. Le salarié en informe l'employeur auparavant ; si l'employeur souhaite également se faire assister, il en informe à son tour le salarié.</p><p>L'employeur peut se faire assister par une personne de son choix appartenant au personnel de l'entreprise ou, dans les entreprises de moins de cinquante salariés, par une personne appartenant à son organisation syndicale d'employeurs ou par un autre employeur relevant de la même branche.</p>"
  },
  "L1237-13": {
    texte: `La convention de rupture définit les conditions de celle-ci, notamment le montant de l'indemnité spécifique de rupture conventionnelle qui ne peut pas être inférieur à celui de l'indemnité prévue à l'article L. 1234-9.

Elle fixe la date de rupture du contrat de travail, qui ne peut intervenir avant le lendemain du jour de l'homologation.

A compter de la date de sa signature par les deux parties, chacune d'entre elles dispose d'un délai de quinze jours calendaires pour exercer son droit de rétractation. Ce droit est exercé sous la forme d'une lettre adressée par tout moyen attestant de sa date de réception par l'autre partie.`
  },
  "L1237-14": {
    texte: `A l'issue du délai de rétractation, la partie la plus diligente adresse une demande d'homologation à l'autorité administrative, avec un exemplaire de la convention de rupture. Un arrêté du ministre chargé du travail fixe le modèle de cette demande.

L'autorité administrative dispose d'un délai d'instruction de quinze jours ouvrables, à compter de la réception de la demande, pour s'assurer du respect des conditions prévues à la présente section et de la liberté de consentement des parties.A défaut de notification dans ce délai, l'homologation est réputée acquise et l'autorité administrative est dessaisie.

La validité de la convention est subordonnée à son homologation.

L'homologation ne peut faire l'objet d'un litige distinct de celui relatif à la convention. Tout litige concernant la convention, l'homologation ou le refus d'homologation relève de la compétence du conseil des prud'hommes, à l'exclusion de tout autre recours contentieux ou administratif. Le recours juridictionnel doit être formé, à peine d'irrecevabilité, avant l'expiration d'un délai de douze mois à compter de la date d'homologation de la convention. `
  },
  "L1237-15": {
    texte: `Les salariés bénéficiant d'une protection mentionnés aux articles L. 2411-1 et L. 2411-2 peuvent bénéficier des dispositions de la présente section. Par dérogation aux dispositions de l'article L. 1237-14, la rupture conventionnelle est soumise à l'autorisation de l'inspecteur du travail dans les conditions prévues au chapitre Ier du titre Ier du livre IV, à la section 1 du chapitre Ier et au chapitre II du titre II du livre IV de la deuxième partie. Dans ce cas, et par dérogation aux dispositions de l'article L. 1237-13, la rupture du contrat de travail ne peut intervenir que le lendemain du jour de l'autorisation.

Pour les médecins du travail, la rupture conventionnelle est soumise à l'autorisation de l'inspecteur du travail, après avis du médecin inspecteur du travail.`
  },
  "L1237-16": {
    texte: `La présente section n'est pas applicable aux ruptures de contrats de travail résultant :

1° Des accords issus de la négociation mentionnée aux articles L. 2242-20 et L. 2242-21 ;

2° Des plans de sauvegarde de l'emploi dans les conditions définies par l'article L. 1233-61 ;

3° Des accords collectifs mentionnés à l'article L. 1237-17.`
  },
  "R1237-3": {
    texte: `L'autorité administrative compétente pour l'homologation de la convention de rupture prévue à l'article L. 1237-14 est le directeur régional des entreprises, de la concurrence, de la consommation, du travail et de l'emploi du lieu où est établi l'employeur.
NOTA :

Décret n° 2009-1377 du 10 novembre 2009 article 7 I : Les dispositions du présent décret prennent effet, dans chaque région, à la date de nomination du directeur régional des entreprises, de la concurrence, de la consommation, du travail et de l'emploi (Les arrêtés de nomination de ces directeurs ont été publiés par arrêtés des 30 décembre 2009 et 9 février 2010, parus respectivement au Journal officiel des 5 janvier et 14 février 2010).

Conformément à l'article 15 du même décret elles ne s'appliquent ni à la région Ile-de-France ni aux régions d'outre-mer.

Conformément à l'article 2 du décret n° 2010-687 du 24 juin 2010, les dispositions du décret n° 2009-1377 du 10 novembre 2009 s'appliquent à la région Ile-de-France à compter du 1er juillet 2010.`
  },
  //Démission
  "L1237-1": {
    texte: `En cas de démission, l'existence et la durée du préavis sont fixées par la loi, ou par convention ou accord collectif de travail.

En l'absence de dispositions légales, de convention ou accord collectif de travail relatifs au préavis, son existence et sa durée résultent des usages pratiqués dans la localité et dans la profession.

Un décret en Conseil d'Etat détermine les modalités d'application du présent article.`
  },
  //Principes de la rupture
  "L1231-1": {
    texte: `Le contrat de travail à durée indéterminée peut être rompu à l'initiative de l'employeur ou du salarié, ou d'un commun accord, dans les conditions prévues par les dispositions du présent titre.

Ces dispositions ne sont pas applicables pendant la période d'essai.`
  },
  "L1231-2": {
    texte: `Les dispositions du présent titre ne dérogent pas aux dispositions légales assurant une protection particulière à certains salariés.`
  },
  "L1231-3": {
    texte: `Les dispositions du présent titre sont applicables lorsque le salarié est lié à plusieurs employeurs par des contrats de travail.`
  },
  "L1234-9": {
    texte: `Le salarié titulaire d'un contrat de travail à durée indéterminée, licencié alors qu'il compte 8 mois d'ancienneté ininterrompus au service du même employeur, a droit, sauf en cas de faute grave, à une indemnité de licenciement.

Les modalités de calcul de cette indemnité sont fonction de la rémunération brute dont le salarié bénéficiait antérieurement à la rupture du contrat de travail. Ce taux et ces modalités sont déterminés par voie réglementaire.`
  }
};
