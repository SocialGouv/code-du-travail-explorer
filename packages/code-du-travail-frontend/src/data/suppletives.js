const suppletives = `

6100  R3141-4; L3141-11
6420  L3141-16;
6200  L3141-23;
5300  L3142-5;
4800  D3142-5; D3142-6; L3142-15;
4600  D3142-11; D3142-12; D3142-13; L3142-27;
5400  D3142-20; D3142-19; D3142-21; L3142-34; L3142-35;
5000  R3142-28; L3142-41; R3142-29
4500  L3142-47; D3142-32;
5100  L3142-53; D3142-35
4400  D3142-43; R3142-44;
4700  L3142-66; R3142-52; R3142-53;
4900  D3142-56;D3142-57;
4200  L3142-78;
5200  D3142-73; D3142-74; D3142-75; D3142-76; L3142-119; L3142-120; L3142-121; L3142-122; L3142-123; L3142-124;
21700 L3121-8;
19800 L3121-12; R3121-3;
20500 L3121-15
20200 L3121-23; L3121-24; L3121-25; L3121-26; R3121-11; R3121-12; R3121-13; R3121-14; R3121-15; R3121-16;
20350 L3121-35;
20360 L3121-36; L3121-37
20370 L3121-38;L3121-39;L3121-40;D3121-18;D3121-19;D3121-20; D3121-21; D3121-22;D3121-23;D3121-24;
19700 D3121-27; D3121-28; L3121-45; L3121-46; L3121-47
19600 L3121-52; R3121-30; R3121-34; R3121-35;
20100 L3121-65; L3121-66;
21600 L3122-20; L3122-21; L3122-22; L3122-23; L3122-24; R3122-9; R3122-10;
5700  L3153-1; L3153-2; D3154-1; D3154-2; D3154-3; D3154-4; D3154-5; D3154-6;
21100 L3131-3; D3131-7;
21500 L3123-26; L3123-28; L3123-29; L3123-30; L3123-31; L3123-27; D3123-2; D3123-3;
20700 L3133-3-2;
20700 L3133-12;

`;

// to clean dict
const dispos = suppletives
  .split("\n")
  .filter(Boolean)
  .reduce((a, c) => {
    const matches = c.match(/^(\d+)\s+(.*)$/);
    const id = parseInt(matches[1]);
    const articles = matches[2]
      .split(";")
      .filter(Boolean)
      .map(x => x.trim());
    a[id] = articles;
    return a;
  }, {});

export default dispos;
