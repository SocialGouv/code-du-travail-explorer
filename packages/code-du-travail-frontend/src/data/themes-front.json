../../../code-du-travail-data/dataset/themes-front.json