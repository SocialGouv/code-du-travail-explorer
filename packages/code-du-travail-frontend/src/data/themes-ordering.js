// Custom order and groups for items found in any level of the breadcrumb path.
// The aim is to make information more human readable in the UI.
//
// This file is tightly coupled to the names used in `themes.js` and must be
// kept in sync.
//
// Each key of the `ordering` object must be the current path of the breadcrumb
// and each value must be an array of 1 or more arrays that group items, e.g.:
//
//   let ordering = {
//     "Contrat de travail > Droit Disciplinaire": [
//       [
//         "Contrôle juridictionnel",
//         "Pénal",
//       ],
//       [
//         "Procédure disciplinaire: pénal",
//         "Procédure disciplinaire: prescription",
//       ],
//     ],
//   }

let ordering = {
  // The `root` path is represented as an empty string in the breadcrumb path.
  // So the empty string is used as an object identifier on purpose.
  "": [
    ["Contrat de travail", "Durée du travail et congés", "Rémunération"],
    ["Pouvoirs de l'employeur, libertés et droits fondamentaux"],
    ["Emploi - Formation"],
    ["Institutions représentatives du personnel", "Négociations collectives"],
    ["Santé Sécurité"],
    ["Obligations diverses"]
  ],

  "Contrat de travail": [
    [
      "Embauche",
      "généralités",
      "Contrat: exécution, modification, transfert",
      "Maladie, Accident du travail (AT), Inaptitude",
      "Maternité, Paternité",
      "Droit Disciplinaire"
    ],
    ["Période d'essai", "Rupture de contrat à durée Indéterminée (CDI)"],
    [
      "Professions particulières",
      "Règles spécifiques du Contrat à durée déterminée (CDD)",
      "Règles spécifiques du Contrat de travail temporaire",
      "Travail temporaire et employeur public"
    ],
    ["Salariés étrangers et déplacés", "Travailleurs étrangers ou détachés"],
    ["Conseil de prud'hommes (CPH)", "Autres cas de mise à disposition"]
  ],

  "Durée du travail et congés": [
    ["Durée du travail"],
    ["Congés payés", "Compte épargne temps", "Congés autres"],
    ["Spécificités jeunes"]
  ],

  "Durée du travail et congés > Durée du travail": [
    [
      'Durée légale "35 heures"',
      "Décompte de la durée du travail",
      "Durée maximale du travail",
      "Jours fériés",
      "Repos et temps de pause"
    ],
    ["Heures supplémentaires", "Aménagement Temps de Travail", "Forfait"],
    ["Temps partiel", "Travail intermittent"],
    [
      "Astreintes, travail effectif et équivalences",
      "Travail de nuit",
      "Travail du dimanche"
    ],
    ["Outre - mer"]
  ],

  "Contrat de travail > Rupture de contrat à durée Indéterminée (CDI)": [
    [
      "Principes de la rupture",
      "Licenciement",
      "Rupture conventionnelle (individuelle)",
      "Rupture à l'initiative du salarié",
      "Rupture d'un commun accord prévue par un accord collectif"
    ],
    ["Cas particuliers", "Retraite"]
  ],

  "Contrat de travail > Période d'essai": [
    [
      "Définition",
      "Une clause nécessaire du contrat de travail",
      "Généralités"
    ],
    [
      "Durée maximale par catégories de salariés",
      "Exceptions relatives à la durée",
      "Renouvellements : possibilités et durée"
    ],
    [
      "Délai de prévenance et Rupture par le salarié",
      "Délai de prévenance et Rupture par l'employeur"
    ],
    ["stagiaire"]
  ]
};

export default ordering;
