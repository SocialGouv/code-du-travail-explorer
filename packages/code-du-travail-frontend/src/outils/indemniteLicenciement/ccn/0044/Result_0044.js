import React from "react";
import MathJax from "react-mathjax-preview";
import { ErrorBoundary } from "../../../../common/ErrorBoundary";
import { Container, Button } from "@cdt/ui";
import {
  getIndemnite as getIndemniteConventionnelle,
  getSalaireRef as getSalaireRefConventionnel
} from "./0044_indemnite";
import { getIndemnite, getSalaireRef } from "../../indemnite";
import { SectionTitle, Highlight, SmallText } from "../../stepStyles";

function Result_Chimie({ form }) {
  const state = form.getState();

  const {
    hasTempsPartiel = false,
    hasSameSalaire = false,
    inaptitude = false,
    salairePeriods = [],
    salaires = [],
    primes = [],
    salaire,
    anciennete,
    dateNotification,
    isEco,
    hasOpe,
    age,
    groupe
  } = state.values;

  const salaireRefLegal = getSalaireRef({
    hasTempsPartiel,
    hasSameSalaire,
    salaire,
    salairePeriods,
    salaires,
    anciennete,
    primes
  });

  const { indemnite } = getIndemnite({
    salaireRef: salaireRefLegal,
    anciennete,
    inaptitude,
    dateNotification
  });

  const salaireRef = getSalaireRefConventionnel({
    salaireRefLegal,
    groupe,
    salaires,
    salaire,
    dateNotification
  });
  const { indemniteConventionnelle, formula } = getIndemniteConventionnelle({
    salaireRef,
    indemnite,
    anciennete,
    isEco,
    hasOpe,
    age,
    groupe
  });

  return (
    <Container>
      <SectionTitle>Indemnité Conventionnelle</SectionTitle>
      <p>
        Le montant de l’indemnité est{" "}
        <Highlight>{indemniteConventionnelle} €</Highlight>{" "}
        <SmallText>
          {indemniteConventionnelle > indemnite
            ? "sur la base du calcul de l'indemnité conventionelle"
            : "sur la base du calcul de l’indemnité légale"}
        </SmallText>
      </p>

      <br />
      <details>
        <summary>Voir le detail du calcul</summary>
        <ErrorBoundary>
          <MathJax math={"`" + formula + "`"} />
        </ErrorBoundary>
      </details>
      <br />
      <br />
      <Button> Recommencer une simulation </Button>
    </Container>
  );
}

export default Result_Chimie;
