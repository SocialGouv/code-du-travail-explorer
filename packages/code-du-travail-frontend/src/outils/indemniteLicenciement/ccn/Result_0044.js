import React from "react";
import PropTypes from "prop-types";
import { Container, Section } from "@cdt/ui";
import MathJax from "react-mathjax2";
import { headStyle } from "../stepStyles";
const round = fl => parseFloat((fl * 100) / 100).toFixed(2);

const labelize = value => {
  if (isNaN(value) || !value) {
    return "-";
  }
  return `${round(value)} €`;
};

const Row = ({ children, value }) => (
  <tr>
    <td style={{ textAlign: "left", fontWeight: "bold" }}>{children}</td>
    <td style={{ textAlign: "center" }}>{value}</td>
  </tr>
);
Row.propTypes = {
  children: PropTypes.node,
  value: PropTypes.string
};

class ResultDetail extends React.Component {
  static propTypes = {
    formula: PropTypes.string.isRequired,
    salaireRef: PropTypes.number.isRequired,
    indemnite: PropTypes.number.isRequired,
    age: PropTypes.number.isRequired,
    echelon: PropTypes.shape({
      groupe: PropTypes.string,
      coefficient: PropTypes.string
    }).isRequired,
    hasOpe: PropTypes.bool.isRequired,
    isEco: PropTypes.bool.isRequired
  };

  render() {
    const {
      formula,
      salaireRef,
      moyenneSalaires,
      dernierSalaire,
      indemnite,
      convention,
      age,
      anciennete,
      echelon,
      hasOpe
    } = this.props;
    return (
      <React.Fragment>
        <Section light>
          <div style={headStyle}>
            <h2>Montant indicatif de votre indemnité</h2>
            <h3 style={{ fontSize: "2rem" }}>{labelize(indemnite)}</h3>
            <p>
              <em>Calculé sur la base de la </em>
              {convention.ccName}
            </p>
          </div>
        </Section>
        <Container>
          <table width="100%" style={{ fontSize: "1.2em" }}>
            <tbody>
              <Row value={`${labelize(moyenneSalaires)}`}>
                Moyenne des 12 derniers mois
              </Row>
              <Row value={`${labelize(dernierSalaire)}`}>
                Dernier salaire
                {echelon.groupe === "V" && "avant préavis"}
              </Row>
              <Row value={`${labelize(salaireRef)}`}>Salaire retenu</Row>
              <Row value={echelon.groupe}>Echelon</Row>
              <Row value={hasOpe ? "Oui" : "non"}>
                L&apos;entreprise est affiliée a une Organisation Patronnale
                Employeur
              </Row>
              <Row value={`${age}`}>Age</Row>
              <Row value={`${anciennete / 12}`}>
                Ancienneté en mois par année
              </Row>
            </tbody>
          </table>
        </Container>
        <Section light>
          <MathJax.Context input="ascii">
            <div
              style={{
                fontSize: "1.5em",
                textAlign: "center",
                fontFamily: "MJXc-TeX-main-R,MJXc-TeX-main-Rw"
              }}
            >
              <MathJax.Node inline>{formula}</MathJax.Node>;
            </div>
          </MathJax.Context>
        </Section>
      </React.Fragment>
    );
  }
}

export default ResultDetail;
