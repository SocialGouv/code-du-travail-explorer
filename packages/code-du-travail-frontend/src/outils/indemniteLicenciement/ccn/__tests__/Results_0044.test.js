import React from "react";
import { render } from "react-testing-library";
import Result_0044 from "../Result_0044";
import { Form } from "react-final-form";
import data from "../../__tests__/form.mock.json";

const form = {
  getState: () => data
};

describe("<Result_0044 />", () => {
  it("should render", () => {
    const onSubmit = jest.fn();
    const { container } = render(
      <Form onSubmit={onSubmit} render={() => <Result_0044 form={form} />} />
    );
    expect(container).toMatchSnapshot();
  });
});
