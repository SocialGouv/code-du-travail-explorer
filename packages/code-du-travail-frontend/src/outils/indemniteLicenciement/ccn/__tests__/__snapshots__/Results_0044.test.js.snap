// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Result_0044 /> should render 1`] = `
.c0 {
  max-width: 1200px;
  margin: 0 auto;
  padding: 0 1.25rem;
}

.c0 > *:last-child {
  margin-bottom: 0;
}

.c1 {
  padding: 0.625rem 1rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  text-align: center;
  line-height: inherit;
  font-size: 1rem;
  font-weight: 600;
  vertical-align: middle;
  border-style: solid;
  border-width: 1px 1px 2px 1px;
  border-radius: 0.25rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms ease;
  transition: background-color 250ms ease;
  color: #fff;
  background: #006be6;
  border-color: #006be6;
  border-bottom-color: #0053b3;
}

.c1:not([disabled]):hover,
.c1:not([disabled]):focus {
  background: #07f;
  color: #fff;
}

.c1:not([disabled]):active {
  color: #fff;
  background: #1a85ff;
  border-width: 2px 1px 1px 1px;
  border-color: #006be6;
  outline: none;
}

.c1[aria-pressed="true"] {
  color: #fff;
  background: #07f;
  border-width: 2px 1px 1px 1px;
  border-color: #006be6;
  border-top-color: #0053b3;
  box-shadow: inset 0 1px 2px 0 #0053b3;
}

.c1[aria-pressed="true"]:not([disabled]):hover,
.c1[aria-pressed="true"]:not([disabled]):focus {
  background-color: #006be6;
}

.c1[aria-pressed="true"]:not([disabled]):active {
  border-top-color: #0053b3;
}

.c1[disabled] {
  cursor: not-allowed;
  color: rgba(255,255,255,0.4);
}

@media print {
  .c0 {
    max-width: 100%;
    padding: 0;
  }
}

<div>
  <div
    class="c0"
  >
    <h2>
      Indemnité Conventionnelle
    </h2>
    <p>
      Le montant de l’indemnité conventionnelle est supérieur à celui de l’indemnité légale. 
      <br />
       Le montant de votre indémnité de licenciement sera de 
      <strong>
        1500
         €
      </strong>
    </p>
    <br />
    <details>
      <summary>
        Voir le detail du calcul
      </summary>
      <div
        id="react-mathjax-preview"
      >
        <div
          id="react-mathjax-preview-result"
        >
          \`3/10 * 2000 * 2.5 \`
        </div>
      </div>
    </details>
    <br />
    <br />
    <button
      aria-pressed="false"
      class="c1"
    >
       Recommencer une simulation 
    </button>
  </div>
</div>
`;
